tests/integration/server/semester/semester.service.test.ts

```diff
@@ -377,6 +383,62 @@ describe('Semester Service', function () {
         }
       });
     });
+    describe('faculty', function () {
+      const [lastSpringSemester] = semesters.slice(-1);
+      const testAcademicYear = lastSpringSemester.academicYear + 1;
+      describe(`is ${ABSENCE_TYPE.NO_LONGER_ACTIVE}`, function () {
+        it('carries over from previous available academic year', async function () {
+          const { id } = await facultyRepository
+            .createQueryBuilder('f')
+            .leftJoinAndSelect('f.absences', 'absence')
+            .getOne();
+
+          await absenceRepository.createQueryBuilder()
+            .update(Absence)
+            .set({ type: ABSENCE_TYPE.NO_LONGER_ACTIVE })
+            .where('facultyId = :id', { id })
+            .execute();
+
+          await semesterService.addAcademicYear(testAcademicYear);
+
+          const { absences } = await facultyRepository
+            .createQueryBuilder('f')
+            .leftJoinAndSelect('f.absences', 'absence')
+            .leftJoinAndSelect('absence.semester', 'semester')
+            .where({ id })
+            .andWhere(
+              'semester.academicYear = :acyr',
+              { acyr: testAcademicYear }
+            )
```

Comment:
`semester.academicYear` is actually the calendar year while the `addAcademicYear` function is creating Spring in `year` and Fall in `year - 1`. 
So assuming that `testAcademicYear` is 2024, this would create Fall 2023 and Spring 2024, but the `WHERE` clause here would only return the "Spring 2024" semester.

I'd suggest duplicating the test format used above, and query `absenceService` to get the specific spring and fall absences associated with the faculty member. Then you can have separate assertions for each absences that check if `type === ABSENCE_TYPE.NO_LONGER_ACTIVE`